src/zkey/cca.c

```diff
@@ -181,13 +220,16 @@ int load_cca_library(struct cca_lib *cca, bool verbose)
 	    cca->dll_CSUACRD == NULL ||
 	    cca->dll_CSNBKTB2 == NULL ||
 	    cca->dll_CSNBKTR2 == NULL ||
-	    cca->dll_CSNBRKA == NULL) {
+	    cca->dll_CSNBRKA == NULL ||
+	    cca->dll_CSNDPKB == NULL ||
+	    cca->dll_CSNDPKG == NULL ||
+	    cca->dll_CSNDPKX == NULL ||
+	    cca->dll_CSNDPKI == NULL ||
+	    cca->dll_CSNDKTC == NULL) {
 		pr_verbose(verbose, "%s", dlerror());
 		DEBUG("The command requires the IBM CCA Host Libraries and "
 		      "Tools.\nFor the supported environments and downloads, "
 		      "see:\n%s", CCA_WEB_PAGE);
-		dlclose(cca->lib_csulcca);
-		cca->lib_csulcca = NULL;
```

Comment:
Why did you remove this? I would keep it that way (also see comment above).
If a certain function is optional, then don't check that one here, but check all the others.

Reply:
this would make all the checks for individual functions being NULL in the functions bleow.

Reply:
see comment above.